airflow/utils/task_group.py

```diff
@@ -453,6 +454,21 @@ def topological_sort(self, _include_subdag_tasks: bool = False):
 
         return graph_sorted
 
+    def get_task_dict(self) -> dict[str, AbstractOperator]:
+        """Returns a flat dictionary of task_id: AbstractOperator"""
+        task_map = {}
+
+        def build_map(dag_node):
+            if not isinstance(dag_node, TaskGroup):
+                task_map[dag_node.task_id] = dag_node
+                return
+
+            for child in dag_node.children.values():
+                build_map(child)
```

Comment:
It’s probably a good idea to rewrite this as a loop instead of recursive calls, we have had some issues with very deep graphs in the past.

Reply:
done